### PR TITLE
Cache

### PR DESCRIPTION
**Pull request by kateinoigakukun**
IBLinter checks whole files every time but there are many unnecessary processes when most of them are not changed.

This PR adds lint cache for each IB file.
In my project, IBLinter took 7 sec every time but this change reduced it to 2 sec when there is cache.

### CODE REVIEW

File: Sources/IBLinterKit/Config/Config.swift [hunk 1]
```diff
@@ -74,6 +79,7 @@ public struct Config: Codable {
         viewAsDeviceRule = try container.decodeIfPresent(Optional<ViewAsDeviceConfig>.self, forKey: .viewAsDeviceRule) ?? nil
         reporter = try container.decodeIfPresent(String.self, forKey: .reporter) ?? "xcode"
         disableWhileBuildingForIB = try container.decodeIfPresent(Bool.self, forKey: .disableWhileBuildingForIB) ?? true
+        ignoreCache = try container.decodeIfPresent(Bool.self, forKey: .ignoreCache) ?? false
```
Comment (kateinoigakukun) [inline Sources/IBLinterKit/Config/Config.swift]: What do you think about default value? Should we enable cache by default?

Comment (phimage) [reply Sources/IBLinterKit/Config/Config.swift]: Yes enabled by default is the good choice if there is no way to forgot some issues

**Comment by kateinoigakukun** [comment]
@phimage Your understanding is almost correct. As you said, we should re-check if IBLinter version changed.
So I use the following values to re-check files.
- IBLinter version https://github.com/IBDecodable/IBLinter/pull/124/files#diff-c687cb12e1ee55a2364f871dd9c8caa5R33
- .iblinter.yml file hash https://github.com/IBDecodable/IBLinter/pull/124/files#diff-c687cb12e1ee55a2364f871dd9c8caa5R83
- modification date https://github.com/IBDecodable/IBLinter/pull/124/files#diff-c687cb12e1ee55a2364f871dd9c8caa5R65

If the top two are changed, IBLinter ignore existing cache files and re-check all files again.

**Comment by kateinoigakukun** [comment]
@phimage Thank you for reviewing 🦄

**Pull request merged**